Maps/MapReader.cpp

```diff
@@ -63,28 +62,18 @@ namespace MapReader {
 			streamReader.SeekRelative(0x1E025);
 		}
 
-		void ReadHeader(StreamReader& streamReader, MapData& mapData)
-		{
-			streamReader.Read(mapData.header);
-		}
-
 		void ReadTiles(StreamReader& streamReader, MapData& mapData)
 		{
 			mapData.tiles.resize(mapData.header.TileCount());
 			streamReader.Read(&mapData.tiles[0], mapData.tiles.size() * sizeof(TileData));
 		}
 
-		void ReadClipRect(StreamReader& streamReader, ClipRect& clipRect)
-		{
-			streamReader.Read(clipRect);
-		}
-
 		void ReadTilesetHeader(StreamReader& streamReader)
 		{
-			char buffer[10];
-			streamReader.Read(buffer, sizeof(buffer));
+			std::array<char, 10> buffer;
+			streamReader.Read(buffer);
 
-			if (std::strncmp(buffer, "TILE SET\x1a", sizeof(buffer))) {
+			if (buffer != std::array<char, 10>{ "TILE SET\x1a" }) {
```

Comment:
I like using `array<char, 10>` here, definitely cleaner.

Maybe we should declare a global/static `const` for the `std::array<char, 10>{ "TILE SET\x1a" }` at the top of the .cpp file. That should move the initialization cost from run time to compile time. Basically the same as what was done for the VOL and CLM tags.

Reply:
Sounds good.